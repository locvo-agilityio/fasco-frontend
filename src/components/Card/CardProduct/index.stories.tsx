import type { Meta, StoryObj } from '@storybook/react';

// Components
import CardProduct from '.';

const meta = {
  title: 'Components/Card/CardProduct',
  component: CardProduct,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof CardProduct>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    img: 'https://down-vn.img.susercontent.com/file/vn-11134207-7ras8-m3sbmx26g0cc16.webp',
    title: 'Sofa',
    priceSale: 5000,
    price: 9550,
    id: '1',
  },
};
