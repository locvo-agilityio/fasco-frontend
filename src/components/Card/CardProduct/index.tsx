'use client';

import { memo } from 'react';
import Image from 'next/image';
import { Card, CardBody, CardFooter, Chip } from '@heroui/react';
import Link from 'next/link';
import isEqual from 'react-fast-compare';

// Utils
import { formatNumber } from '@/utils';

// Constants
import { ROUTES } from '@/constants';

interface CardProductProps {
  id: string;
  img: string;
  title: string;
  price: number;
  priceSale?: number;
  colors?: { name: string; code: string }[];
}

const CardProduct = ({
  id,
  img,
  title,
  price,
  priceSale = 0,
  colors,
}: CardProductProps) => (
  <Card
    as={Link}
    href={ROUTES.PRODUCT_DETAIL(id)}
    shadow="none"
    className="w-[302px] rounded-none hover:shadow-2xl hover:cursor-pointer"
  >
    <CardBody className="flex items-center w-full h-80">
      <Image
        alt={title}
        fill
        src={img}
        priority
        sizes="50vw"
        style={{ objectFit: 'fill' }}
      />
    </CardBody>
    <CardFooter className="w-full p-5 flex-col items-start text-small justify-between">
      <div className="w-full flex flex-col justify-between gap-2">
        <b className="text-black font-volkhov text-base font-normal truncate">
          {title}
        </b>
        {priceSale ? (
          <div className="flex gap-2">
            <b className="text-black font-jost text-base font-normal">
              ${formatNumber(priceSale)}
            </b>
            <b className="text-gray-500 font-jost text-base font-normal line-through">
              ${formatNumber(price)}
            </b>
          </div>
        ) : (
          <b className="text-black font-jost text-base font-normal">
            ${formatNumber(price)}
          </b>
        )}
      </div>

      <div className="flex gap-2 mt-3">
        {colors?.map((color) => (
          <Chip
            key={color.code}
            className="p-0 w-6 h-6 min-w-6 border-2"
            style={{ backgroundColor: color.code }}
          />
        ))}
      </div>
    </CardFooter>
  </Card>
);

export default memo(CardProduct, isEqual);
