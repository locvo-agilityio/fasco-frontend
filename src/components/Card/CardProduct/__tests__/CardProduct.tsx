import { render } from '@testing-library/react';

// Components
import CardProduct from '..';

const props = {
  img: 'https://m.media-amazon.com/images/I/51JpA4Olu4L._AC_UL1157_.jpg',
  title: 'Sofa',
  price: 9550,
  priceSale: 5000,
  brand: 'Al Karam',
  rating: 5,
  reviews: 4100,
  id: '1',
  colors: [{ name: 'Black', code: '#000000' }],
};

describe('CardProduct', () => {
  it('should render correctly', () => {
    const { container } = render(<CardProduct {...props} />);

    expect(container).toMatchSnapshot();
  });
});
