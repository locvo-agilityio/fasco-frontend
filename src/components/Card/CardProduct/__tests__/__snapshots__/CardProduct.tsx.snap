// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProduct should render correctly 1`] = `
<div>
  <a
    class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-none transition-transform-background motion-reduce:transition-none w-[302px] rounded-none hover:shadow-2xl hover:cursor-pointer"
    href="/products/1"
    tabindex="-1"
  >
    <div
      class="relative p-3 flex-auto flex-col place-content-inherit align-items-inherit break-words text-left overflow-y-auto subpixel-antialiased flex items-center w-full h-80"
    >
      <img
        alt="Sofa"
        data-nimg="fill"
        decoding="async"
        sizes="50vw"
        src="/_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FI%2F51JpA4Olu4L._AC_UL1157_.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: fill; color: transparent;"
      />
    </div>
    <div
      class="h-auto flex overflow-hidden color-inherit subpixel-antialiased rounded-b-large w-full p-5 flex-col items-start text-small justify-between"
    >
      <div
        class="w-full flex flex-col justify-between gap-2"
      >
        <b
          class="text-black font-volkhov text-base font-normal truncate"
        >
          Sofa
        </b>
        <div
          class="flex gap-2"
        >
          <b
            class="text-black font-jost text-base font-normal"
          >
            $
            5k
          </b>
          <b
            class="text-gray-500 font-jost text-base font-normal line-through"
          >
            $
            9.6k
          </b>
        </div>
      </div>
      <div
        class="flex gap-2 mt-3"
      >
        <div
          class="relative max-w-fit inline-flex items-center justify-between box-border whitespace-nowrap text-small rounded-full bg-default text-default-foreground p-0 w-6 h-6 min-w-6 border-2"
          style="background-color: rgb(0, 0, 0);"
        >
          <span
            class="flex-1 text-inherit font-normal px-2"
          />
        </div>
      </div>
    </div>
  </a>
</div>
`;
